Gateway/Request/AddressHandlerPool.php

```diff
@@ -58,7 +58,8 @@ public function __construct(array $addressHandlers)
     public function getShippingAddress(Order $order)
     {
         try {
-            $shippingAddress = clone $order->getShippingAddress();
+            $orderShippingAddress = $order->getShippingAddress() ?? $order->getBillingAddress();
+            $shippingAddress = clone $orderShippingAddress;
```

Comment:
why do you clone this object?

Reply:
Because I don't want to change the shippingAddress from the magento table. I need the shippingAddress to be updated only in the Pay Request